requirements-hyperopt.txt

```diff
@@ -6,3 +6,6 @@ scipy==1.15.2
 scikit-learn==1.6.1
 ft-scikit-optimize==0.9.2
 filelock==3.18.0
+optuna==4.2.1
+optunahub==0.2.0
```

Comment:
optunahub doesn't seem to be used? it also doesn't seem to be something freqtrade actually needs ? 

cmaes ... is that an optional dependency for optuna?

Reply:
My first intention was to use AutoSampler which need optunahub (https://hub.optuna.org/samplers/auto_sampler/), but for some reason cannot be pickled. Need to investigate more why this is happening. 
So I changed the default to optuna.samplers.NSGAIISampler.
For now these are the choices for samplers: "TPESampler","GPSampler","CmaEsSampler","NSGAIISampler","NSGAIIISampler","QMCSampler"

cmaes  is dependency for optuna.samplers.CmaEsSampler.
Not sure how if all samplers should be available ....

Reply:
well the short answer is - pickle is a pain :laughing: 

you can't pickle anything that has an open socket, nothing containing a Lock object, . . . (there's a few more limitations, really) :laughing:

Reply:
That is very true.
Autotune It is working with latest version from git - they have a fix (https://github.com/optuna/optunahub-registry/commit/e0afdd82c12b285056f6d3118aedc200a42d34a7), but I guess this is not yet in the pip package.
Anyhow, let's see if with default NSGAIISampler works ok. If yes, the optunahub and AutoSampler can be added later

Reply:
Later - or via [`generate_estimator`](https://www.freqtrade.io/en/latest/advanced-hyperopt/#overriding-base-estimator) ... ;)

Reply:
removed optunahub dependency. Autotune can be used using via [generate_estimator]. Now default is NSGAIIISampler (even is experimental) which gives best result comparing to NSGAIISampler or TPESampler for strategies I tested.